### PR TITLE
blog: APISIX Wasm support

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendu@apache.org>

New blog post titled "How Does Apache APISIX Support WebAssembly?"

### CODE REVIEW

File: content/posts/apisix-wasm-support.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+Wasm binaries are executed in Wasm virtual machines (VMs). They implement the WASI standard, allowing Wasm binaries to be run outside the browser.
+
+The wasm-nginx-module implements two Wasm VMs, [Wasmtime](https://wasmtime.dev/) and [WasmEdge](https://wasmedge.org/), and uses the former as default.
+
+Running Wasm plugins inside a VM comes with beneficial side effects. It offers security and isolation from APISIX, ensuring that any issues with the plugin are self-contained.
+
+The wasm-nginx-module first [loads](https://github.com/api7/wasm-nginx-module/blob/ccc83f7397c711b5f99a55682134a0972a5a6040/lib/resty/proxy-wasm.lua#L81) the `.wasm` binary to the VM and then uses the `call` method, as we saw above. All Wasm plugins are run inside the same Wasm VM.
+
+## Programming Model
```
Comment (Boburmirzo) [inline content/posts/apisix-wasm-support.md]: Maybe it is better to put the Programming Model section after the Proxy Wasm Specification. It makes the topic easier to follow in order.

File: content/posts/apisix-wasm-support.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  image: "/images/apisix-wasm-support/binary-banner.jpeg"
+  alt: "Zeroes and ones."
+  caption: "It's all zeroes and ones if you look deep enough."
+  relative: false
+---
+
+[WebAssembly (Wasm)](https://webassembly.org/) has found its way outside the browser and has been widely adopted on the server side as in browsers with the development of [WebAssembly System Interface (WASI)](https://github.com/WebAssembly/WASI).
+
+In previous articles, we looked at how you can [build plugins for Apache APISIX in languages like Go using Wasm](/posts/tiny-apisix-plugin/) and how you can [optimize its performance](/posts/tinier-apisix-plugin/). We made it pretty clear that Wasm is here to stay in APISIX.
+
```
Comment (Boburmirzo) [inline content/posts/apisix-wasm-support.md:20]: Maybe you summarize and mention all the topics that are being covered in the article here. For example, _this article walks you through how the proxy Wasm spec is implemented, what is Wasm Module for Nginx, which programming models APISIX uses and how Wasm binaries are executed._

**Review by Boburmirzo — APPROVED** [review]
Other than two comments I left, the article looks nice. Thanks!

File: content/posts/apisix-wasm-support.md (hunk 1 above)
Comment (pottekkat) [reply content/posts/apisix-wasm-support.md]: That was the plan but I mentioned Wasm VMs and haven't explained what they were. Maybe I can move the Wasm VMs to the beginning and make the move you suggested.

File: content/posts/apisix-wasm-support.md (hunk 2 above)
Comment (pottekkat) [reply content/posts/apisix-wasm-support.md:20]: In the conclusion?

Comment (Boburmirzo) [reply content/posts/apisix-wasm-support.md:20]: No, it is preferrable at the beginning of article as an agenda to highlight each section.  You mentioned this:

_But how does APISIX support these Wasm plugins? Do they have limitations? What will they look like in the future? In this "under the hood of APISIX" article, we will attempt to answer these questions and more._

However, you wrote only about limitations and future vision only in the summary.

File: content/posts/apisix-wasm-support.md (hunk 1 above)
Comment (Boburmirzo) [reply content/posts/apisix-wasm-support.md]: Yes, this approach is good!

**Pull request merged**